server/src/main/java/org/elasticsearch/index/seqno/ReplicationTracker.java

```diff
@@ -283,6 +283,29 @@ public synchronized RetentionLease renewRetentionLease(final String id, final lo
         return retentionLease;
     }
 
+    /**
+     * Removes an existing retention lease.
+     *
+     * @param id       the identifier of the retention lease
+     * @param listener the callback when the retention lease is successfully removed and synced to replicas
+     */
+    public void removeRetentionLease(final String id, final ActionListener<ReplicationResponse> listener) {
+        Objects.requireNonNull(listener);
+        final RetentionLeases currentRetentionLeases;
+        synchronized (this) {
+            assert primaryMode;
+            if (retentionLeases.contains(id) == false) {
+                throw new IllegalArgumentException("retention lease with ID [" + id + "] does not exist");
```

Comment:
This will change to a `RetentionLeaseNotFoundException` with the integration of #38754.